packages/@ember/-internals/glimmer/lib/helpers/unique-id.ts

```diff
@@ -30,7 +30,12 @@ import { createConstRef } from '@glimmer/reference';
 import { internalHelper } from './internal-helper';
 
 export default internalHelper((): Reference<string> => {
-  return createConstRef(uniqueId(), 'unique-id');
+  // SAFETY: glimmer-vm should change the signature of createUnboundRef to use a generic
+  //         so that the type param to `Reference<?>` can infer from the first argument.
+  //
+  //         However, since ember-source@4.5, we can use functions explicitly,
+  //         so we should investigate if we even need this "internalHelper" wrapper.
```

Comment:
It's more the other way around – making const references is the optimization, using the user-land plain function helpers facilities would be creating extra wrappers, tracking frames, etc

Reply:
Resolved, thanks for explaining!